### ISSUE TITLE
xud grpc unresponsive (seemingly because of TOR connection attempts)

### ISSUE THREAD

**Issue by kilrau** [ExchangeUnion/xud-docker#241]
On xud1 and xud1.testnet after restarting xud, the xud grpc needs 2-4 minutes until it's responding (throwing `Error: 14 UNAVAILABLE: failed to connect to all addresses` aka "not ready yet" until then).

xud log:
```
xud_1        | 04/12/2019 16:25:14.475 [GLOBAL] info: Local nodePubKey is 037a74737630bb5a9979762b2163ef9f95c81e00caf347bacc17f16991cdbb61f7
xud_1        | 04/12/2019 16:25:14.521 [P2P] info: Connecting to testnet XU network
xud_1        | 04/12/2019 16:25:14.527 [P2P] info: p2p server listening on 0.0.0.0:18885
xud_1        | 04/12/2019 16:25:14.530 [P2P] debug: Verifying reachability of advertised address: pso32ae2l5nyh3clgee2nj2hgpmcnluhaqibjm2r2fjcsn4gclxfmrid.onion:18885
xud_1        | 04/12/2019 16:25:14.534 [P2P] debug: Verifying reachability of advertised address: xud1.testnet.exchangeunion.com:18885
xud_1        | 04/12/2019 16:25:14.537 [HTTP] info: http server listening on 0.0.0.0:8887
xud_1        | 04/12/2019 16:25:18.523 [P2P] debug: Connected pre-handshake to 127.0.0.1:32976
xud_1        | 04/12/2019 16:25:18.526 [P2P] debug: Connected pre-handshake to 23.129.64.182:35112
xud_1        | 04/12/2019 16:25:18.592 [RPC] info: gRPC server listening on 0.0.0.0:8886
xud_1        | 04/12/2019 16:25:18.595 [RAIDEN] error: unexpected raiden status 404: NOT FOUND
xud_1        | 04/12/2019 16:25:18.596 [RAIDEN] error: failed to update maximum outbound capacity: 5.6 - unexpected error during raiden request
xud_1        | 04/12/2019 16:25:18.602 [RAIDEN] error: unexpected raiden status 404: NOT FOUND
xud_1        | 04/12/2019 16:25:18.635 [P2P] info: Connecting to known / previously connected peers
xud_1        | Dec 04 16:25:25.000 [notice] Closed 1 streams for service [scrubbed].onion for reason resolve failed. Fetch status: No more HSDir available to query.
xud_1        | Dec 04 16:25:33.000 [notice] Closed 1 streams for service [scrubbed].onion for reason resolve failed. Fetch status: No more HSDir available to query.
xud_1        | Dec 04 16:25:41.000 [notice] Closed 1 streams for service [scrubbed].onion for reason resolve failed. Fetch status: No more HSDir available to query.
xud_1        | Dec 04 16:25:48.000 [notice] Closed 1 streams for service [scrubbed].onion for reason resolve failed. Fetch status: No more HSDir available to query.
xud_1        | Dec 04 16:26:11.000 [notice] Closed 1 streams for service [scrubbed].onion for reason resolve failed. Fetch status: No more HSDir available to query.
xud_1        | Dec 04 16:26:18.000 [notice] Closed 1 streams for service [scrubbed].onion for reason resolve failed. Fetch status: No more HSDir available to query.
xud_1        | Dec 04 16:26:25.000 [notice] Closed 1 streams for service [scrubbed].onion for reason resolve failed. Fetch status: No more HSDir available to query.
xud_1        | Dec 04 16:26:37.000 [notice] Closed 1 streams for service [scrubbed].onion for reason resolve failed. Fetch status: No more HSDir available to query.
xud_1        | Dec 04 16:26:44.000 [notice] Closed 1 streams for service [scrubbed].onion for reason resolve failed. Fetch status: No more HSDir available to query.
xud_1        | Dec 04 16:26:54.000 [notice] Closed 1 streams for service [scrubbed].onion for reason resolve failed. Fetch status: No more HSDir available to query.
xud_1        | Dec 04 16:27:00.000 [notice] Closed 1 streams for service [scrubbed].onion for reason resolve failed. Fetch status: No more HSDir available to query.
xud_1        | Dec 04 16:27:06.000 [notice] Closed 1 streams for service [scrubbed].onion for reason resolve failed. Fetch status: No more HSDir available to query.
xud_1        | Dec 04 16:27:13.000 [notice] Closed 1 streams for service [scrubbed].onion for reason resolve failed. Fetch status: No more HSDir available to query.
xud_1        | Dec 04 16:27:23.000 [notice] Closed 1 streams for service [scrubbed].onion for reason resolve failed. Fetch status: No more HSDir available to query.
xud_1        | Dec 04 16:27:30.000 [notice] Closed 1 streams for service [scrubbed].onion for reason resolve failed. Fetch status: No more HSDir available to query.
xud_1        | Dec 04 16:27:37.000 [notice] Closed 1 streams for service [scrubbed].onion for reason resolve failed. Fetch status: No more HSDir available to query.
xud_1        | Dec 04 16:27:46.000 [notice] Closed 1 streams for service [scrubbed].onion for reason resolve failed. Fetch status: No more HSDir available to query.
xud_1        | 04/12/2019 16:27:56.307 [P2P] error: Peer timed out waiting for response to packet 0
xud_1        | 04/12/2019 16:27:56.308 [P2P] debug: Peer (127.0.0.1:32976): closing socket. reason: ResponseStalling
xud_1        | 04/12/2019 16:27:56.309 [P2P] error: Peer timed out waiting for response to packet 0
xud_1        | 04/12/2019 16:27:56.309 [P2P] debug: Peer (23.129.64.182:35112): closing socket. reason: ResponseStalling
xud_1        | 04/12/2019 16:27:56.312 [P2P] warn: could not open connection to peer (127.0.0.1:32976): response timeout (SessionInit) 
xud_1        | 04/12/2019 16:27:56.313 [P2P] warn: could not open connection to peer (23.129.64.182:35112): response timeout (SessionInit) 
xud_1        | 04/12/2019 16:27:56.313 [P2P] warn: Could not verify reachability of advertised address: pso32ae2l5nyh3clgee2nj2hgpmcnluhaqibjm2r2fjcsn4gclxfmrid.onion:18885
xud_1        | 04/12/2019 16:27:56.313 [P2P] warn: Could not verify reachability of advertised address: xud1.testnet.exchangeunion.com:18885
xud_1        | 04/12/2019 16:27:56.314 [P2P] warn: could not open connection to peer (020bfbd1fcd158fd9031f5fd971a2402fbf2d698d5f3afe1fb5eaa2803dd643882@co63noo7g7iu2zhhzj7r5f6rnn7z456ufmjo4haavc4bhvyxufownoyd.onion:18885): could not connect to peer at co63noo7g7iu2zhhzj7r5f6rnn7z456ufmjo4haavc4bhvyxufownoyd.onion:18885: connect ECONNREFUSED 224.0.0.3:18885
xud_1        | 04/12/2019 16:27:56.315 [P2P] warn: could not open connection to peer (03cf92fd60ef014b679ad22f1d74842da6b1445a92719434ee7889728ee9215f51@ndg65ec5d3ptz5sxlfby4n6luq55vdghv3ercilorsymbwfwlzbvepyd.onion:18885): could not connect to peer at ndg65ec5d3ptz5sxlfby4n6luq55vdghv3ercilorsymbwfwlzbvepyd.onion:18885: connect ECONNREFUSED 224.0.0.4:18885
xud_1        | 04/12/2019 16:27:56.315 [P2P] warn: could not open connection to peer (02a7f31324a12ad8a0c1ac5202072fc9a354b31a7de5b238a703ec452ae9307ec4@ge27cetaajruglnclzwcih2vml3bxkpe2voxk4pbsajc5xv3ltwxfdqd.onion:18885): could not connect to peer at ge27cetaajruglnclzwcih2vml3bxkpe2voxk4pbsajc5xv3ltwxfdqd.onion:18885: connect ECONNREFUSED 224.0.0.5:18885
xud_1        | 04/12/2019 16:27:56.316 [P2P] warn: could not open connection to peer (02393f313808c6a8cbf4a9d53e98c5f128ea6deeda7ab085f10de774bec67557e2@me64yg2sq5tv67agfz7rubedbczar4tl2mrs3zg56xlvbvkpntpg2hyd.onion:18885): could not connect to peer at me64yg2sq5tv67agfz7rubedbczar4tl2mrs3zg56xlvbvkpntpg2hyd.onion:18885: connect ECONNREFUSED 224.0.0.6:18885
xud_1        | 04/12/2019 16:27:56.316 [P2P] warn: could not open connection to peer (031ce2a23911dc9f1cdc0986fa86c4ecc604d2fb7f34cad94b71446f24a6d7c458@5ubb7fph6zqi4l2vazpha7rj62kv6i7vbvbsxtn57oyju3lxrmxwctad.onion:18885): could not connect to peer at 5ubb7fph6zqi4l2vazpha7rj62kv6i7vbvbsxtn57oyju3lxrmxwctad.onion:18885: connect ECONNREFUSED 224.0.0.7:18885
xud_1        | 04/12/2019 16:27:56.318 [P2P] warn: could not open connection to peer (020cfcfc5d4dc4801f0a81aa35030bd381c71fb74afc7f8dbb137a09820bbe6690@kjts3yytf4hstizorebhhdezm7phh4pusgphxvhb53gaabe2463vcfyd.onion:18885): could not connect to peer at kjts3yytf4hstizorebhhdezm7phh4pusgphxvhb53gaabe2463vcfyd.onion:18885: connect ECONNREFUSED 224.0.0.8:18885
xud_1        | 04/12/2019 16:27:56.318 [P2P] warn: could not open connection to peer (022a4f42be78c1e81f9ba2f42d1821fb34ef4383c26591c444b021bd57b54461ba@olikel3cwmadblyq5ykcx2vkmqxaoqtjeif2yrjn5vxavi72yxi7kxqd.onion:18885): could not connect to peer at olikel3cwmadblyq5ykcx2vkmqxaoqtjeif2yrjn5vxavi72yxi7kxqd.onion:18885: connect ECONNREFUSED 224.0.0.9:18885
xud_1        | 04/12/2019 16:27:56.319 [P2P] warn: could not open connection to peer (02b18d755c2ae4da02c7b6dd06a22cb834db3edee0dc3c3212b8df5aae313b6fd3@u7wibll7lbwrazasoua2arm6qo46muxek7rh4vpsd2vo5xkstvnhuzad.onion:18885): could not connect to peer at u7wibll7lbwrazasoua2arm6qo46muxek7rh4vpsd2vo5xkstvnhuzad.onion:18885: connect ECONNREFUSED 224.0.0.10:18885
xud_1        | 04/12/2019 16:27:56.319 [P2P] warn: could not open connection to peer (0258d625a3805d2119a87032e7331980b3e7fa2db641f836309616a1f95711347d@sq5t6fplhr5ubk7cthiewwg6zpaaaejjbr2vyl6ht3ivdajyhlix2jqd.onion:18885): could not connect to peer at sq5t6fplhr5ubk7cthiewwg6zpaaaejjbr2vyl6ht3ivdajyhlix2jqd.onion:18885: connect ECONNREFUSED 224.0.0.11:18885
xud_1        | 04/12/2019 16:27:56.320 [P2P] warn: could not open connection to peer (0237583f55a7d02560930d67e9ac698f61b2d3b73918f8b20617510f1b5f5b3b02@bxs3hykhrbd7avy3wzmz6afo2hrxqw6deambyqrjq3swtzhawlr4djqd.onion:18885): could not connect to peer at bxs3hykhrbd7avy3wzmz6afo2hrxqw6deambyqrjq3swtzhawlr4djqd.onion:18885: connect ECONNREFUSED 224.0.0.12:18885
xud_1        | 04/12/2019 16:27:56.320 [P2P] warn: could not open connection to peer (02992de59e291eba4734164392535be37662d0655a789782040fba534ae6b7f227@5hr6mzjpd4zsjrcm3f2rrra5ixbrrkw6eafs5ckjfv6ng5fdmjbftsqd.onion:18885): could not connect to peer at 5hr6mzjpd4zsjrcm3f2rrra5ixbrrkw6eafs5ckjfv6ng5fdmjbftsqd.onion:18885: connect ECONNREFUSED 224.0.0.13:18885
xud_1        | 04/12/2019 16:27:56.320 [P2P] warn: could not open connection to peer (023c4f62aa48912b149034f8294659060474c7275e1252e07a70c58f51a12b95ba@5ugx5yk7bwsh2qsafe6ctwyi6uldlwy4dnz5f2a7mft5c3ef4j7zlvqd.onion:18885): could not connect to peer at 5ugx5yk7bwsh2qsafe6ctwyi6uldlwy4dnz5f2a7mft5c3ef4j7zlvqd.onion:18885: connect ECONNREFUSED 224.0.0.14:18885
xud_1        | 04/12/2019 16:27:56.321 [P2P] warn: could not open connection to peer (0268d7a595f6fafd0376a420aa034f15f625bd682cdfe408903874cb7a62918877@eq72k4k56suwullwqxg537f2x2wafrgok7fmivqlbb73arzyd543y3qd.onion:18885): could not connect to peer at eq72k4k56suwullwqxg537f2x2wafrgok7fmivqlbb73arzyd543y3qd.onion:18885: connect ECONNREFUSED 224.0.0.15:18885
xud_1        | 04/12/2019 16:27:56.321 [P2P] warn: could not open connection to peer (03fe0cd35dd8d0c3ca637d567e631d6557adece2b5b649731c91a857a9c9df1c69@vsjeldt5wtj3sundtk5gpn3glia26aes3l2squhcurffg37z6tjiztyd.onion:18885): could not connect to peer at vsjeldt5wtj3sundtk5gpn3glia26aes3l2squhcurffg37z6tjiztyd.onion:18885: connect ECONNREFUSED 224.0.0.16:18885
xud_1        | 04/12/2019 16:27:56.322 [P2P] warn: could not open connection to peer (029a76ce950d5b3ccfaf6ccccf08ad7139b3fe463fccd10eb4de6cf171b26e0267@wqaihq37p4dcd4jwfzgdnz5w5672egxmx4dgw2polj2eqj5vaekpcryd.onion:18885): could not connect to peer at wqaihq37p4dcd4jwfzgdnz5w5672egxmx4dgw2polj2eqj5vaekpcryd.onion:18885: connect ECONNREFUSED 224.0.0.17:18885
xud_1        | 04/12/2019 16:27:56.322 [P2P] warn: could not open connection to peer (02c16149445f1651eed11ccd0c4debf39d6b239b71016b7c05495c623ee504afbf@suxdqwd7nm7orzlzhpdmzhuf4bzkacx756cp5davqkuq24etb2puw6ad.onion:18885): could not connect to peer at suxdqwd7nm7orzlzhpdmzhuf4bzkacx756cp5davqkuq24etb2puw6ad.onion:18885: connect ECONNREFUSED 224.0.0.18:18885
xud_1        | 04/12/2019 16:27:56.323 [P2P] warn: could not open connection to peer (0224352f0f18c812ccde3f342f82bdac8c6ee225e6745af65cebafc44c450c2b14@hfawesboazirurfmlwwumuquoo7ouu5aqcs7cwi2carlr3j6gzm7h7qd.onion:18885): could not connect to peer at hfawesboazirurfmlwwumuquoo7ouu5aqcs7cwi2carlr3j6gzm7h7qd.onion:18885: connect ECONNREFUSED 224.0.0.19:18885
xud_1        | 04/12/2019 16:27:56.323 [P2P] warn: could not open connection to peer (02f21888d07e160c917f6f26a94d627b958f89919239f01ebb1bd0dafec408354c@lanty67ptc6jeraqhmlzdymiuntzqyflr7y26ijti7t5zrpssjltqyid.onion:18885): could not connect to peer at lanty67ptc6jeraqhmlzdymiuntzqyflr7y26ijti7t5zrpssjltqyid.onion:18885: connect ECONNREFUSED 224.0.0.20:18885
xud_1        | 04/12/2019 16:27:56.323 [P2P] warn: could not open connection to peer (023cc0a31c86c4f4e82634e3f06e2f45558dbc3469d053eac004386cd1a905428f@x6uk3kzq4fuduu4nszckypxelzafb45flsdom63pfbicxitic2xxb4id.onion:18885): could not connect to peer at x6uk3kzq4fuduu4nszckypxelzafb45flsdom63pfbicxitic2xxb4id.onion:18885: connect ECONNREFUSED 224.0.0.22:18885
xud_1        | Dec 04 16:28:02.000 [notice] Closed 1 streams for service [scrubbed].onion for reason resolve failed. Fetch status: No more HSDir available to query.
xud_1        | 04/12/2019 16:28:02.576 [P2P] debug: Connected pre-handshake to 0312c3d5ce9f653a4e3f0e6a0c5491fe6b1a434ba201351b0c10bb857a64bae4e7@43xdm6l6ammsscxa7uprnh4khnieoman37o5fvquwkibmax2tj3glhad.onion:18885
xud_1        | 04/12/2019 16:28:02.585 [P2P] info: Socket closed prior to handshake with 23.129.64.182:35112
xud_1        | 04/12/2019 16:28:02.585 [P2P] info: Socket closed prior to handshake with 127.0.0.1:32976
xud_1        | 04/12/2019 16:28:02.587 [P2P] debug: Connection attempt #1 to 020bfbd1fcd158fd9031f5fd971a2402fbf2d698d5f3afe1fb5eaa2803dd643882@co63noo7g7iu2zhhzj7r5f6rnn7z456ufmjo4haavc4bhvyxufownoyd.onion:18885 failed: connect ECONNREFUSED 224.0.0.3:18885. retrying in 5 sec...
xud_1        | 04/12/2019 16:28:02.590 [P2P] debug: received disconnecting packet from 0312c3d5ce9f653a4e3f0e6a0c5491fe6b1a434ba201351b0c10bb857a64bae4e7@43xdm6l6ammsscxa7uprnh4khnieoman37o5fvquwkibmax2tj3glhad.onion:18885 due to ResponseStalling
xud_1        | 04/12/2019 16:28:02.593 [P2P] info: Socket closed prior to handshake with 0312c3d5ce9f653a4e3f0e6a0c5491fe6b1a434ba201351b0c10bb857a64bae4e7@43xdm6l6ammsscxa7uprnh4khnieoman37o5fvquwkibmax2tj3glhad.onion:18885
xud_1        | 04/12/2019 16:28:02.593 [P2P] warn: could not open connection to peer (0312c3d5ce9f653a4e3f0e6a0c5491fe6b1a434ba201351b0c10bb857a64bae4e7@43xdm6l6ammsscxa7uprnh4khnieoman37o5fvquwkibmax2tj3glhad.onion:18885): Peer 0312c3d5ce9f653a4e3f0e6a0c5491fe6b1a434ba201351b0c10bb857a64bae4e7@43xdm6l6ammsscxa7uprnh4khnieoman37o5fvquwkibmax2tj3glhad.onion:18885 disconnected from us due to ResponseStalling

```

**Comment by kilrau** [comment]
I didn't experience this in the last days. You? @erkarl

**Comment by erkarl** [comment]
Just tested this again on xud1 testnet. 4+ minutes until the gRPC became available after unlocking.

**Comment by erkarl** [comment]
Fixed by #1395.
